src/main/java/org/elasticsearch/index/shard/ShadowIndexShard.java

```diff
@@ -81,28 +83,28 @@ public ShadowIndexShard(ShardId shardId, @IndexSettings Settings indexSettings,
                             IndexService indexService, ShardSuggestService shardSuggestService, ShardQueryCache shardQueryCache,
                             ShardBitsetFilterCache shardBitsetFilterCache, @Nullable IndicesWarmer warmer,
                             SnapshotDeletionPolicy deletionPolicy, SimilarityService similarityService,
-                            MergePolicyProvider mergePolicyProvider, EngineFactory factory, ClusterService clusterService) {
+                            MergePolicyProvider mergePolicyProvider, EngineFactory factory, ClusterService clusterService, NodeEnvironment nodeEnv) {
         super(shardId, indexSettings, indexSettingsService, indicesLifecycle, store, mergeScheduler,
                 translog, threadPool, mapperService, queryParserService, indexCache, indexAliasesService,
                 indexingService, getService, searchService, shardWarmerService, shardFilterCache,
                 shardFieldData, percolatorQueriesRegistry, shardPercolateService, codecService,
                 termVectorsService, indexFieldDataService, indexService, shardSuggestService,
                 shardQueryCache, shardBitsetFilterCache, warmer, deletionPolicy, similarityService,
-                mergePolicyProvider, factory, clusterService);
+                mergePolicyProvider, factory, clusterService, nodeEnv);
     }
 
     /**
      * In addition to the regular accounting done in
-     * {@link IndexShard#routingEntry(org.elasticsearch.cluster.routing.ShardRouting)},
+     * {@link IndexShard#updateRoutingEntry(org.elasticsearch.cluster.routing.ShardRouting, boolean)},
      * if this shadow replica needs to be promoted to a primary, the shard is
      * failed in order to allow a new primary to be re-allocated.
      */
     @Override
-    public IndexShard routingEntry(ShardRouting newRouting) {
+    public void updateRoutingEntry(ShardRouting newRouting, boolean persistState) {
         if (newRouting.primary() == true) {// becoming a primary
             throw new ElasticsearchIllegalStateException("can't promote shard to primary");
         }
-        return super.routingEntry(newRouting);
+        super.updateRoutingEntry(newRouting, persistState);
```

Comment:
OK. I wonder if it's needed as they can be assigned everywhere. But this is definitely a different change (if at all)